TCAT/AppDelegate.swift

```diff
@@ -160,6 +160,79 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         alertController.addAction(UIAlertAction(title: "OK", style: .cancel, handler: nil))
         UIApplication.shared.keyWindow?.presentInApp(alertController)
     }
+
+    /// When app is opened via URL
+    // URLs for testing
+    // BusStop: ithaca-transit://getRoutes?lat=42.442558&long=-76.485336&stopName=Collegetown
+    // PlaceResult: ithaca-transit://getRoutes?lat=42.44707979999999&long=-76.4885196&destinationName=Hans%20Bethe%20House
+    func application(_ app: UIApplication, open url: URL, options: [UIApplication.OpenURLOptionsKey: Any] = [:]) -> Bool {
+
+        let rootVC = HomeViewController()
+        let navigationController = CustomNavigationController(rootViewController: rootVC)
+        self.window = UIWindow(frame: UIScreen.main.bounds)
+        self.window!.rootViewController = navigationController
+        self.window?.makeKeyAndVisible()
+
+        let urlComponents = NSURLComponents(url: url, resolvingAgainstBaseURL: false)
+        let items = (urlComponents?.queryItems)! as [NSURLQueryItem]
+        var stop: BusStop
+
+        // if the URL is from the getRoutes intent
+        if url.absoluteString.contains("getRoutes") {
+            var latitude: CLLocationDegrees?
+            var longitude: CLLocationDegrees?
+            var stopName: String?
+            let optionsVC = RouteOptionsViewController()
+
+            // get parameters from the URL
+            for (index, element) in items.enumerated() {
```

Comment:
how do u know that latitude will always be the "0th" element in the items array? ik you hardcode the query parameters in the following method, but I think it would be better practice to lookup values by their key. E.g. if the query string is `ithaca-transit://getRoutes?lat=42.44707979999999&long=-76.4885196&destinationName=Hans%20Bethe%20House` you'd look up `lat` by doing something like `queryString[lat]`

Maybe try this:
https://stackoverflow.com/questions/8756683/best-way-to-parse-url-string-to-get-values-for-keys